opencensus/stats/view.py

```diff
@@ -33,13 +40,19 @@ class View(object):
     :param aggregation: the aggregation the view will support
 
     """
+
     def __init__(self, name, description, columns, measure, aggregation):
         self._name = name
         self._description = description
         self._columns = columns
         self._measure = measure
         self._aggregation = aggregation
 
+        # Cache the converted MetricDescriptor here to avoid creating it each
+        # time we convert a ViewData that realizes this View into a Metric.
+        self._md_cache_lock = threading.Lock()
```

Comment:
Other stats classes are generally not threadsafe. I don't know that it's worth opening this can of worms in this PR.